lib/ssh_pearl.sh

```diff
@@ -18,49 +18,81 @@
 #
 
-function ssh_pearl(){
-    function _aggregate_scripts(){
-        local fileRc="$1"
-        local dirRc="$2"
-        [[ -f "${fileRc}" ]] && cat "${fileRc}"
-        [[ -d "${dirRc}" ]] && cat "${dirRc}"/*
+
+function ssh_pearl() {
+    local self lib rcScript inputrcScript vimrcScript ssh_opts opt commands arg commands_opt is_script
+
+    commands=()
+    ssh_opts=()
+
+    function _aggregate_scripts() {
+        local node
+
+        echo -e '\n'  # to enforce there's linefeed after previous block
+        for node in "$@"; do
+            [[ -f "$node" ]] && cat "$node"
+            [[ -d "$node" ]] && cat "$node"/*
+        done
     }
+
+    [[ "$1" == __SCRIPT_INVOCATION__ ]] && is_script=1 && shift
     [[ -z "$@" ]] && { ssh; return $?; }
 
-    local ssh_opts=()
     for opt in "$@"; do
         case "$opt" in
             --) shift ; break ;;
-            *) ssh_opts+=("$opt") ; shift ;;
+            *)  ssh_opts+=("$opt") ; shift ;;
         esac
     done
 
-    local commands=()
-    for arg in "$@"; do commands+=("$arg"); done
-    local commands_opt=""
-    [[ -z "${commands[@]}" ]] || commands_opt="-c '${commands[@]}'"
+    for arg in "$@"; do
+        commands+=("$arg")
+    done
+
+    [[ -n "${commands[@]}" ]] && commands_opt="-c '${commands[*]}'"
 
     command -v base64 >/dev/null 2>&1 || { echo >&2 "pearl-ssh requires base64 to be installed locally. Aborting."; return 1; }
     command -v gzip >/dev/null 2>&1 || { echo >&2 "pearl-ssh requires gzip to be installed locally. Aborting."; return 1; }
 
     [[ -z "$PEARL_HOME" ]] && PEARL_HOME=${HOME}/.config/pearl
 
-    local rcScript="$(_aggregate_scripts "$PEARL_HOME/sshrc" "$PEARL_HOME/sshrc.d" | gzip | base64)"
-    local inputrcScript="$(_aggregate_scripts "$PEARL_HOME/sshinputrc" "$PEARL_HOME/sshinputrc.d" | gzip | base64)"
-    local vimrcScript="$(_aggregate_scripts "$PEARL_HOME/sshvimrc" "$PEARL_HOME/sshvimrc.d" | gzip | base64)"
+    [[ "$is_script" -ne 1 && "$__REMOTE_SSH" -ne 1 ]] && rcScript="$(declare -f "$FUNCNAME")"
+    rcScript+="$(_aggregate_scripts "$PEARL_HOME/bashrc" "$PEARL_HOME/bashrc.d")"
+    rcScript="$(echo "$rcScript" | gzip | base64)"
+    inputrcScript="$(_aggregate_scripts "$PEARL_HOME/inputrc" "$PEARL_HOME/inputrc.d" | gzip | base64)"
+    vimrcScript="$(_aggregate_scripts "$PEARL_HOME/vimrc" "$PEARL_HOME/vimrc.d" | gzip | base64)"
+    if [[ "$is_script" -eq 1 ]]; then
+        self="$(gzip < "$0" | base64)"
+        lib="$(gzip < "$(resolve_real_path "$0")/../lib/$LIBNAME" | base64)"
+    fi
 
     CMD="
-        for tmpDir in /tmp \$HOME; do [[ -w \"\$tmpDir\" ]] && { foundTmpDir=\"\$tmpDir\"; break; } done;
-        [[ -z \"\$foundTmpDir\" ]] && { echo >&2 \"Could not find writable tempdirs on the server. Aborting.\"; exit 1; };
+        export __REMOTE_SSH=1  # states we're over ssh session
+        for tmpDir in /tmp \$HOME; do [[ -w \"\$tmpDir\" ]] && { foundTmpDir=\"\$tmpDir\"; break; } done
```

Comment:
Was wondering about the ';' usage, good to know.
But why the reversion? Readability wise it might be better to move the export down to the `PEARLSSH_HOME' export, but functionality wise has no impact whatsoever.